pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1371,6 +1373,12 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
         if (!name.isGlobal()) {
             return CompletableFuture.completedFuture(null);
         }
+        NamespaceName heartbeatNamespace = NamespaceService.getHeartbeatNamespaceV2(
```

Comment:
we  move this into the PulsarService as a field, this way  we don't have to create this every time